sc_cost_meter/utils.py

```
@@ -0,0 +1,185 @@
+import boto3
+import logging
+import os
+
+from datetime import datetime
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+def get_ec2_client():
+  return boto3.client('ec2')
+
+def get_ssm_client():
+  return boto3.client('ssm')
+
+def get_ce_client():
+  return boto3.client('ce')
+
+def get_meteringmarketplace_client():
+  return boto3.client('meteringmarketplace')
+
+def get_dynamo_client():
+  return boto3.client('dynamodb')
+
+def get_env_var_value(env_var):
+  '''Get the value of an environment variable
+  :param env_var: the environment variable
+  :returns: the environment variable's value, None if env var is not found
+  '''
+  value = os.getenv(env_var)
+  if not value:
+    log.warning(f'cannot get environment variable: {env_var}')
+
+  return value
+
+def get_marketplace_synapse_ids():
+  '''Get Synapse IDs from the Marketplace Dynamo DB, these are the Service Catalog customers.
+  Assumes that there is a Dynamo DB with a table containing a mapping of Synapse
+  IDs to SC subscriber data
+  :return a list of synapse IDs, otherwise return empty list if no customers are in DB
+  '''
+  synapse_ids = []
+  ddb_marketplace_table_name = get_env_var_value('MARKETPLACE_ID_DYNAMO_TABLE_NAME')
+  ddb_marketplace_synapse_user_id_attribute = "SynapseUserId"
+  if ddb_marketplace_table_name:
```

When we complete the P.O.C. we need to change this to raise an error if the table is not found.  Otherwise we will silently fail to bill customers for usage.

---

will do.